src/miner.cpp

```diff
@@ -495,9 +495,9 @@ void BlockAssembler::addPackageTxs(int &nPackagesSelected, int &nDescendantsUpda
 
         // Decide if add transaction as alert or regular transaction
         auto addToBlock = [&] (CTxMemPool::txiter entry) {
-            if (alertsEnabled) {
+            if ( auto tx = entry->GetTx(); alertsEnabled && !IsLicenseTx(tx)) {
```

Comment:
Should `auto tx = entry->GetTx();` really be placed as `if` condition?

Reply:
Changed